### PR TITLE
Migrate to Linkify React

### LINKED ISSUE #1961: Migrate from `react-linkify` to `linkify-react`
Look into the implementation of [`linkify-react`](https://linkify.js.org/docs/linkify-react.html). If it satisfies our requirements, migrate to that package. Else, look into building a more robust implementation from our side.
cc: @Amaljith-K

### PR DESCRIPTION
**Pull request by ittyjose**
- Fixes #1961 

**Description**
This PR will migrate the package from `react-linkify` to `linkify-react`
**Checklist**

~- [ ] I have made corresponding changes to the documentation.~
~- [ ] I have updated the types definition of modified exports.~
- [x] I have verified the functionality in some of the neeto web-apps.
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**
@CalvinChiramal _a

### CODE REVIEW

File: src/components/Toastr/Toast.jsx [hunk 1]
```diff
@@ -10,11 +10,11 @@ const Toast = ({ type, message, buttonLabel = "", onClick = noop }) => (
...
-      componentDecorator={(decoratedHref, decoratedText, key) => (
-        <a href={decoratedHref} key={key} rel="noreferrer" target="_blank">
-          {decoratedText}
-        </a>
-      )}
+      options={{
+        defaultProtocol: "https",
+        rel: "noreferrer",
+        target: "_blank",
+      }}
```
Comment (ajmaln) [inline src/components/Toastr/Toast.jsx:17]: Does this also consider `emails` that needs to be displayed as links wich will open a mail client on click?

**Review by ajmaln — COMMENTED** [review]
@ittyjose _a Please check

File: src/components/Toastr/Toast.jsx (hunk 1 above)
Comment (ittyjose) [reply src/components/Toastr/Toast.jsx:17]: Yes. I have verified it and checked the docs too - [link](https://linkify.js.org/docs/linkify-react.html)

**Comment by ittyjose** [comment]
@ajmaln _a

**Review by ajmaln — APPROVED** [review]

**Pull request merged**